src/test/java/com/device/shop/test/controller/ProductControllerTest.java

```diff
@@ -64,6 +77,29 @@ public void testGetAllProducts() throws Exception{
                 .andExpect(jsonPath("$[1].description").value("*"))
                 .andExpect(jsonPath("$[1].sku").value("one"));
     }
+
+    @Test
+    public void testUploadFile() throws Exception {
+        MultipartFile file = Mockito.mock(MultipartFile.class);
+        when(file.getOriginalFilename()).thenReturn("test.csv");
+
+        ResponseEntity<ResponseMessage> response = productController.uploadFile(file);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals("Uploaded the file successfully: test.csv", response.getBody().getMessage());
+        verify(productService, times(1)).save(file);
+    }
+
+    @Test
+    void save_InvalidCSVFile_ThrowsBadRequestException() {
+        ProductRepository productRepository = mock(ProductRepository.class);
+        ProductService productService1 = new ProductService(productRepository);
+        MultipartFile file = new MockMultipartFile("file", "test.txt", "text/plain", "invalid csv".getBytes());
```

Comment:
This test is for service, not controller. Create a service for controller, not service